flowapi/flowapi/user_model.py

```diff
@@ -88,24 +93,39 @@ def _get_query_kind_and_aggregation_unit(self, query_json: dict) -> Tuple[str, s
 
         Returns
         -------
-        str, str
-            Query kind and aggregation unit
+        list of tuples str, str
+            List of tuples giving query kind and aggregation unit
```

Comment:
Again, note for future refactorings: the return data structure is getting a bit more complicated than calling code should need to know. There is likely a missing object hidden here which should take on some of the responsibility about which permissions exist and how to validate them.